crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -1621,11 +1623,15 @@ impl Backend {
                     log_index += sim_res.logs.len();
                     call_res.push(sim_res);
                 }
+                let transactions_envelopes: Vec<AnyTxEnvelope> = transactions
+                .iter()
+                .map(|tx| AnyTxEnvelope::from(tx.clone()))
+                .collect();
 
                 let header = Header {
-                    logs_bloom: Default::default(),
-                    transactions_root: Default::default(),
-                    receipts_root: Default::default(),
+                    logs_bloom:Default::default(),
+                    transactions_root: calculate_transaction_root(&transactions_envelopes),
+                    receipts_root: calculate_receipt_root(&transactions_envelopes.clone()),
```

Comment:
this one isn't quite right

this requires a bit more prep, because we need first need to collect all `Receipt` objects in that block:

https://github.com/alloy-rs/alloy/blob/def933930b5fac1b3278477cbe9c01f99b7a21db/crates/consensus/src/receipt/receipts.rs#L16-L16

then create the WithBloom variant:

https://github.com/alloy-rs/alloy/blob/def933930b5fac1b3278477cbe9c01f99b7a21db/crates/consensus/src/receipt/receipts.rs#L51-L51

and then we can finally calculate the receipts root

and for the `logs_bloom` field we can take a look at the `alloy_primitives::logs_bloom` fn